### DISCUSSION TITLE
Windows Terminal v1.21.2361.0

### DISCUSSION THREAD

**Discussion by PankajBhojwani** [microsoft/terminal#17808] [Announcements]
Terminal 1.21 is finally stable! It brings in a crazy number of improvements, and we once considered it a "large" release. Oh boy.

Here are the highlights:
* Terminal will now remember what was on your screen when you exit, and can restore it if you'd like!
* You can specify multiple active fonts, which will be used in order.
* Box and line drawing glyphs are now rendered pixel-perfect with appropriate stippled shading
* We've rewritten the IME (input method editor) integration to make life much easier for those of you who need to input Japanese, Chinese, Korean, Vietnamese and more
* Scrollbar marks are finally generally available!
* We will now show and highlight all search results at the same time when you search (thansk @tusharsnx!)
* The old text rendering engine has been removed.

Please see the following release notes for additional details.

- [Windows Terminal Preview v1.21.1772.0](https://github.com/microsoft/terminal/releases/tag/v1.21.1772.0)
- [Windows Terminal Preview v1.21.1382.0](https://github.com/microsoft/terminal/releases/tag/v1.21.1382.0)
- [Windows Terminal Preview v1.21.1272.0](https://github.com/microsoft/terminal/releases/tag/v1.21.1272.0)

> [!NOTE]
> This version began rolling out to the Dev, Canary and Beta Windows Insider channels when this GitHub Release was created. Other channels will be released
> as our reliability numbers indicate that we haven't broken anything. If you want to update to this release directly, download it below!

### Why are there so many files? How do I choose?

Please visit our page documenting the different [Windows Terminal Distributions]!

We've also backported the following changes from 1.22 Preview.

## Changes

- We have brought the new Windows 24H2 API out to the ConPTY NuGet package (#17704)
- We've reintroduced support for <kbd>Alt+Numpad</kbd> input of extended off-keyboard characters (#17637) (#17774)

### UI

- Due to a build issue, we had to disable full optimization for this build. We apologize for any potential performance regressions. (#17765)
- Terminal now comes with shiny new icons for the Visual Studio Dev Shells (#17706)
- We have restored the `...`s in some command palette names to help users find them (#17715)

### VT Features

- Reports generated in response to VT requests are once again given priority over other input (#17786)

## Bug Fixes

- We will no longer conveniently forget all the colors on the last line of the screen when you resize it (#17568)
- Terminal will now more reliably save your session when your computer reboots to update (#17714)
- Background colors will no longer bleed over the edges of the screen when D2D is in use (such as over RDP to a VM) (#17674)
- The rendering of curly underlines has been improved (#17501) (thanks @smprather for reporting!) (#17508)
- You can once again cancel overwriting a key binding in the Settings UI (#17763)
- You can once again disable the detection of URLs (#17731)
- You can once again use <kbd>Ctrl+D</kbd> and <kbd>Enter</kbd> in a disconnected pane (#17750)

### Accessibility

- Accessibility API `RangeFromPoint` and `ExpandToEnclosingUnit` will no longer cause Terminal to hang (#17695)

### Reliability

- AtlasEngine will no longer over-read a memory buffer when setting up a GPU texture (#17536)
- Closing a window when you have an `unfocusedAppearance` set will no longer result in unscheduled disassembly of your terminal (#17770)
- We've fixed an instance of Terminal crashing during settings reload (#17751)

### Code Health

- We added `reloadEnvironmentVariables` to our oft-forgotten JSON schema (#17696)
[Windows Terminal Distributions]: https://learn.microsoft.com/windows/terminal/distributions

---

_Binary files inside the unpackaged distribution archive bear the version number `1.21.240823001`._

### Asset Hashes

- Microsoft.WindowsTerminal_1.21.2361.0_8wekyb3d8bbwe.msixbundle_Windows10_PreinstallKit.zip
   - SHA256 `0A5E2048474E11BF4A006543F64AD18E600A1E79878DFEE132BC81F79C28703C`
- Microsoft.WindowsTerminal_1.21.2361.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `19CBE2E77CC814A5AB68E346B0D3D928F44F1EC5A78FB9B53A793D46B1CE3D27`
- Microsoft.WindowsTerminal_1.21.2361.0_arm64.zip
   - SHA256 `E81D085297D8CCCB6AC3CE5E3189E4C63DEEC32EAD444725F8D6245A8C7581A4`
- Microsoft.WindowsTerminal_1.21.2361.0_x64.zip
   - SHA256 `AC2D324EA1AF30CB97D6FB40EF83EBC82E92FBBB516CD274E4ED9CB8FD22FE4E`
- Microsoft.WindowsTerminal_1.21.2361.0_x86.zip
   - SHA256 `B8E3BE7936975665BC3BC5784B03E7799E246D0C99928BD3C4343628EB760A70`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.21.2361.0'>Windows Terminal v1.21.2361.0</a>.</em>

**Comment by mcexit** [comment]
You forgot to redirect the schema URL <https://aka.ms/terminal-profiles-schema> to <https://raw.githubusercontent.com/microsoft/terminal/release-1.21/doc/cascadia/profiles.schema.json>.

You should not be using a redirect for the schema either, as users on older versions will get incompatible versions. Even if they set it manually, it gets replaced whenever a change is made from the GUI.

**Reply by seii188** [reply]
"$ref": "#/$defs/Color",
            "description": "Sets the color used as ANSI green."
          },
          "purple": {
            "$ref": "#/$defs/Color",
            "description": "Sets the color used as ANSI purple."
          },
          "red": {
            "$ref": "#/$defs/Color",
            "description": "Sets the color used as ANSI red."
          },
          "selectionBackground": {
            "$ref": "#/$defs/Color",
            "description": "Sets the selection background color of the color scheme."
          },
          "white": {
            "$ref": "#/$defs/Color",
            "description": "Sets the color used as ANSI white."
          },
          "yellow": {
            "$ref": "#/$defs/Color",
            "description": "Sets the color used as ANSI yellow."
          }
        }
      },
      "type": "array"
    }
  },
  "allOf": [
    {
      "$ref": "#/$defs/Globals"
    },
    {
      "type": "object",
      "additionalItems": true,
      "properties": {
        "profiles": {
          "oneOf": [
            {
              "$ref": "#/$defs/ProfileList"
            },
            {
              "$ref": "#/$defs/ProfilesObject"
            }
          ]
        },
        "schemes": {
          "$ref": "#/$defs/SchemeList"
        }
      },
      "required": [
        "profiles",
        "schemes",
        "defaultProfile"
      ]
    }
  ]
